### PR TITLE
Generalize RL framework to multiple computed params

### PR DESCRIPTION
**Pull request by cpaniaguam**
This PR generalizes the RLDM (Reinforcement Learning Drift-Diffusion Model) likelihood builder to support multiple computed parameters.

#### 1. **Generalized Parameter Computation** (builder.py)
- Dynamically compute any number of parameters listed in `ssm_logp_func.computed`
- Maintain correct parameter ordering by iterating through `ssm_logp_func.inputs`

#### 2. **New Validation System**
- Add `_validate_computed_parameters()` function to catch configuration errors early
- Validate that all computed parameters have corresponding compute functions in `ssm_logp_func.computed`
- Provide clear, actionable error messages instead of confusing downstream failures
- Prevent `ValueError` from `index()` when parameters are misspecified

#### 3. **Enhanced Type Safety**
- Introduced `AnnotatedFunction` Protocol for type-checking functions with metadata
- Added `ColumnLookupResult` dataclass for structured return values
- Improved function signatures with proper type hints throughout

#### 4. **Comprehensive Test Suite** (test_rl_likelihood_builder.py)
- Add 7 new validation tests in `TestValidateComputedParameters` class
- Add 5 new multi-parameter tests in `TestMultipleComputedParameters` class covering:
  - Multiple computed parameters
  - Parameter ordering verification
  - Different input requirements per computed parameter
  - Edge cases (no computed params, all params computed)
- Improved test organization with named tuples (`ModelConfig`, `RLDMData`, `ParamArrays`, `RLDMSetup`)
- **100% code coverage** on builder.py (109/109 statements)

#### 4. **Other Improvements**
- Refactored accumulator patterns to list comprehensions with helper functions
- Enhanced docstrings throughout with NumPy-style documentation
- Added comprehensive examples in `AnnotatedFunction` protocol documentation
- Cleaned up test file with reduced verbosity (650 lines, down from 759)

### Example Usage

```python
# Define multiple computed parameters
@annotate_function(
    inputs=["rl.alpha", "scaler", "response", "feedback"],
    outputs=["v"]
)
def compute_v_subject_wise(subj_trials):
    # RL-based drift rate computation
    ...

@annotate_function(
    inputs=["feedback"],
    outputs=["a"]
)
def compute_a_subject_wise(subj_trials):
    # Arousal-based threshold computation
    ...

# SSM function with multiple computed params
@annotate_function(
    inputs=["v", "a", "z", "t", "theta", "rt", "response"],
    computed={"v": compute_v_subject_wise, "a": compute_a_subject_wise}
)
def angle_logp_func(params_matrix):
    ...

# Builder automatically handles all computed parameters
logp_fn = make_rl_logp_func(
    angle_logp_func,
    n_participants=20,
    n_trials=200,
    list_params=["z", "t", "theta"],  # Only non-computed params
    extra_fields=["feedback"]
)
```

### Testing
- All 19 tests pass (12 existing + 7 new)
- 100% coverage on modified code
- Integration tests verify validation works end-to-end
- PyTensor Op and VJP gradient tests confirm mathematical correctness

### Breaking Changes
None - this is a backward-compatible generalization. Existing code with single computed parameter `v` continues to work unchanged.

### CODE REVIEW

File: src/hssm/rl/likelihoods/builder.py [hunk 1]
```diff
@@ -301,6 +301,46 @@ def _extract_array(source, idx):
...
+            f"in data_cols, list_params, or extra_fields, but no compute "
+            f"functions are provided in ssm_logp_func.computed"
+        )
+
+    missing_compute_funcs = set(computed_params) - set(ssm_logp_func.computed.keys())
+    if missing_compute_funcs:
+        raise ValueError(
+            f"Parameters {sorted(missing_compute_funcs)} are identified as "
+            f"computed but no compute functions are provided in "
+            f"ssm_logp_func.computed"
```
Comment (digicosmos86) [inline src/hssm/rl/likelihoods/builder.py]: These two strings do not need to be f-strings. Surprised that a linter did not catch this

File: src/hssm/rl/likelihoods/builder.py [hunk 2]
```diff
@@ -301,6 +301,46 @@ def _extract_array(source, idx):
...
+        return
+
+    if not hasattr(ssm_logp_func, "computed") or not ssm_logp_func.computed:
+        raise ValueError(
+            f"Parameters {computed_params} are not available "
+            f"in data_cols, list_params, or extra_fields, but no compute "
+            f"functions are provided in ssm_logp_func.computed"
+        )
+
+    missing_compute_funcs = set(computed_params) - set(ssm_logp_func.computed.keys())
```
Comment (digicosmos86) [inline src/hssm/rl/likelihoods/builder.py]: This makes a lot of sense but not very computationally efficient. How about `missing_compute_funcs = [param for param in set(computed_params) if param not in ssm_logp_func.computed]`?

Mainly, I just don't like the creation of a set from the keys of a dict since it's already supposed to be a set

File: src/hssm/rl/likelihoods/builder.py [hunk 3]
```diff
@@ -364,39 +429,53 @@ def logp(data, *args) -> np.ndarray:
...
         )
-        ssm_logp_input_indices = _get_column_indices(
-            ssm_logp_func.computed["v"].inputs, data_cols, list_params, extra_fields
+
+        # Validate that all computed parameters have compute functions
+        _validate_computed_parameters(ssm_logp_func, ssm_logp_func_colidxs.computed)
+
+        computed_param_values = (
+            {
+                param_name: compute_parameter(param_name, data, args)[1]
```
Comment (digicosmos86) [inline src/hssm/rl/likelihoods/builder.py]: Is the first return value of `compute_parameter` used somewhere else? If not, is it easier to just return one value?

File: src/hssm/rl/likelihoods/builder.py [hunk 4]
```diff
@@ -343,7 +383,31 @@ def make_rl_logp_func(
...
+        ssm_logp_data = _collect_cols_arrays(data, args, ssm_logp_input_indices)
+
+        # Reshape into 3D array (n_participants, n_trials, n_inputs)
+        # to enable vmapping over participants
+        subj_trials = jnp.stack(ssm_logp_data, axis=1)
+        subj_trials = subj_trials.reshape(n_participants, n_trials, -1)
+
+        # Apply vmap to compute parameter values for all participants
+        vmapped_func = jax.vmap(compute_func, in_axes=0)
+        computed_values = vmapped_func(subj_trials)
```
Comment (digicosmos86) [inline src/hssm/rl/likelihoods/builder.py]: This `vmap` is called every time `compute_parameter` is called. The `vmap` step should be done outside this function

File: tests/rl/test_rl_likelihood_builder.py [hunk 5]
```diff
@@ -1,5 +1,5 @@
 from pathlib import Path
-from typing import NamedTuple
+from typing import NamedTuple, Callable
```
Comment (digicosmos86) [inline tests/rl/test_rl_likelihood_builder.py:2]: `typing.Callable` is deprecated in Python 3.9. Let's use `collections.abc.Callable` instead

File: src/hssm/rl/likelihoods/builder.py (hunk 2 above)
Comment (cpaniaguam) [reply src/hssm/rl/likelihoods/builder.py]: It seems like the key object supports set operations; this works:
```python
missing_compute_funcs = set(computed_params) - ssm_logp_func.computed.keys()
